nix/build.sh

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+
+set -e
+
+# cleanup for artefacts created during builfs
+function cleanup() {
+  trap - EXIT ERR INT QUIT
+
+  if [ -n "$nixResultPath" ]; then
+    echo "Deleting derivations from Nix store..."
+    releaseDrv=$(nix-instantiate --quiet $nixOpts)
+    if [ -n "$releaseDrv" ]; then
+      local releaseSrcPath=$(nix-store -q --binding src $releaseDrv)
+      local releaseOutPath=$(nix-store -q --outputs $releaseDrv)
+      local releaseRefs=( $(nix-store -q --references $releaseDrv) )
+      local prodBuildDrv=$(printf -- '%s\n' "${releaseRefs[@]}" | grep -e "prod-build-android.drv")
+      local prodBuildSrcPath=$(nix-store -q --binding src $prodBuildDrv)
+      local prodBuildOutPath=$(nix-store -q --outputs $prodBuildDrv)
+      nix-store --delete $prodBuildDrv $prodBuildSrcPath $prodBuildOutPath $releaseDrv $releaseSrcPath $releaseOutPath 2> /dev/null
+    fi
+  fi
+}
+
+trap "cleanup" EXIT ERR INT QUIT
+
+function extractResults() {
+  nixResultsPath="$1"
+  mkdir -p results
+  cp -r "${nixResultPath}/*" results/
+  chmod u+w results/
+}
+
+nixOpts=(
+  "--pure"
+  "--fallback"
+  "--no-out-link"
+  "--show-trace"
+  "--argstr target-os ${TARGET_OS}"
+)
+
+if [[ -n "${BUILD_TYPE}" ]]; then
+    nixOpts+=("--argstr build-type ${BUILD_TYPE}")
+fi
+
+targetAttr="${1}"
+shift
+
+if [[ -z "${targetAttr}" ]]; then
+    echo "First argument is madatory and has to specify the Nix attribute!"
```

Comment:
There are 2 indents instead of one here.

Reply:
k